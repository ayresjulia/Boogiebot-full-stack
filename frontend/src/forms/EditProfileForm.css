.Form {
	margin-top: 0px;
	margin-left: auto;
	margin-right: auto;
	display: grid;
	grid-template-columns: 30% 50%;
	text-align: center;
	justify-content: space-evenly;
	width: 100%;
	border-left: 1px solid rgb(238, 238, 238);
	height: auto;
}

.Form-left {
	margin: 15px;
	transition: all .2s ease-in-out;
	text-decoration: none;
	color: rgb(53, 52, 52);
	background-color: rgba(255, 87, 51, 0.5);
	font-size: 1.5rem;
	display: inline-block;
	text-align: center;
	margin-top: 30%;
	margin-bottom: 200px;
}

.Form-right {
	margin-top: 20px;
}

.Form-update {
	font-size: 80px;
}

.Form-profile-pic {
	width: 150px;
	height: 150px;
	border-radius: 50%;
	animation: rotation 0.5s 1 linear;
	animation-delay: 1s;
}

@keyframes rotation {
	from {
		transform: rotate(0deg);
	}
	to {
		transform: rotate(359deg);
	}
}
