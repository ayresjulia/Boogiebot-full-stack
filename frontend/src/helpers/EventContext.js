import React from "react";

/** Context: provides events object and setter for it throughout app. */

const EventContext = React.createContext();

export default EventContext;
