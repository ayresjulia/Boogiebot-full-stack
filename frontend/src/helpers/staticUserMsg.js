const staticMsg = {
	FORM_TIP: "tip.",
	EDIT_PROFILE_FORM_TIP_DESC:
		"while updating your info, keep in mind, that picking a great playlist is key for any party!",
	EDIT_PROFILE_FORM_CONFIRM_PWD: "Confirm password to make changes",
	FORM_USERNAME: "Username",
	FORM_PWD: "Password",
	FORM_FIRST_NAME: "First Name",
	FORM_LAST_NAME: "Last Name",
	FORM_EMAIL: "Email",
	FORM_PROFILE_URL: "Profile URL",
	EVENT_FORM_TIP_DESC:
		"scoop your ice-cream into a cupcake pan and pop in the freezer for a hassle-free	serving at your party.",
	LOGO_NAME: "BoogieBot",
	NAV_MY_EVENTS: "My Events",
	NAV_INSPIRATION: "Get Inspired",
	NAV_CATERING: "Catering",
	NAV_PROFILE: "Profile",
	NAV_LOG_OUT: "Log Out",
	NAV_LOG_IN: "Log In",
	NAV_SIGN_UP: "Sign Up",
	LOADING: "Loading...",
	CATERING_LOOKUP: "lookup",
	CATERING_FOOTNOTE:
		"check the event you want to save the restaurant to, it will appear in your event details",
	CATERING_INFO: "restaurants nearby with your zip code for catering options.",
	EVENTS_CREATE: "create",
	EVENTS_NEW: "new event.",
	HOME_WELCOME_USER: "Welcome, ",
	NON_USER_MSG: "Please, Log in or Sign up to begin.",
	MY_EVENT_DESC: "Description:",
	MY_EVENT_WHEN: "When:",
	MY_EVENT_TIME: "Time:",
	MY_EVENT_LOCATION: "Location:",
	DEFAULT_EVENT_IMG:
		"https://images.unsplash.com/photo-1532117182044-031e7cd916ee?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
	INSPIRATION_FOOTNOTE:
		"check the event you want to save the picture to, it will appear in your event details",
	INSPIRATION_ADD: "add",
	INSPIRATION_INFO: "pictures to your event moodboard for inspiration.",
	HOME_IMG1:
		"https://images.unsplash.com/photo-1620324446278-be6c7bb3ece5?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
	HOME_IMG2:
		"https://images.unsplash.com/photo-1527275393322-8ddae8bd5de9?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
	HOME_IMG3:
		"https://images.unsplash.com/photo-1525268771113-32d9e9021a97?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
	HOME_IMG4:
		"https://images.unsplash.com/photo-1534119238569-839b7389d552?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=800&q=80",
	CONSOLE_EVENTS_ERROR: "ERROR!!! Events could not be loaded",
	CONSOLE_USER_ERROR: "ERROR!!! currentUser.username could not be loaded",
	MY_EVENT_CAT: "catering",
	CATERING_DEFAULT:
		"https://images.unsplash.com/photo-1557962011-4b36a6f5d921?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1350&q=80",
	CATERING_WEB: "website",
	USER_DEFAULT_URL: "https://supersimple.com/wp-content/uploads/peek-a-boo-800-800-200x200.jpg",
	API_ERROR: "There is an error with getting data from API. Please check connection",
	EDIT_EVENT_FORM_TIP_DESC:
		"don't forget about younger audience, set up a cool kids table at your party",
	EDIT_EVENT_FORM_TITLE: "Title",
	EDIT_EVENT_FORM_DESC: "Description",
	EDIT_EVENT_FORM_EVENT_DATE: "Event Date",
	EDIT_EVENT_FORM_EVENT_TIME: "Event Time",
	EDIT_EVENT_FORM_CITY: "City",
	MY_EVENT_RMRK1: "browse pictures in Get Inspired tab to save them here",
	MY_EVENT_RMRK2: "browse restaurants in Catering tab to save them here",
	EVENTS_RMRK:
		"your events will be shown here. click on event to get details or create a new one",
	TARGET_ERR: "targetEvent could not be loaded",
	EVENT_ALERT: "Please choose an event first."
};

export default staticMsg;
