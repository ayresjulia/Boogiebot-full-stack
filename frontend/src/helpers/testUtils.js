import React from "react";
import UserContext from "./UserContext";

const demoUser = {
	username: "testuser",
	firstName: "testfirst",
	lastName: "testlast",
	email: "test@test.net",
	imgUrl: null
};

const UserProvider = ({ children, currentUser = demoUser }) => (
	<UserContext.Provider value={{ currentUser }}>{children}</UserContext.Provider>
);

export { UserProvider };
