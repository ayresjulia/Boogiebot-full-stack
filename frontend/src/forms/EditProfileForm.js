import React, { useState, useContext, useRef, useEffect, useCallback } from "react";
import { Form, FormGroup, Label, Input, Button } from "reactstrap";
import { useHistory } from "react-router-dom";
import "./EditProfileForm.css";
import BoogieBotApi from "../Api";
import { Row, Col } from "react-bootstrap";
import Alert from "../helpers/Alert";
import staticMsg from "../helpers/staticUserMsg";
import UserContext from "../helpers/UserContext";

/** Form to edit user info in db. */

const EditProfileForm = () => {
	const { currentUser, setCurrentUser } = useContext(UserContext);
	const formDataRef = useRef();

	const INITIAL_STATE = {
		username: currentUser.username,
		firstName: currentUser.firstName,
		lastName: currentUser.lastName,
		email: currentUser.email,
		profileUrl: currentUser.profileUrl,
		password: ""
	};
	const [ formData, setFormData ] = useState(INITIAL_STATE);
	const [ formErrors, setFormErrors ] = useState([]);
	const [ formSuccess, setFormSuccess ] = useState(false);
	const history = useHistory();

	const profile =
		currentUser.profileUrl === "" ? staticMsg.USER_DEFAULT_URL : currentUser.profileUrl;

	useEffect(() => {
		formDataRef.current = INITIAL_STATE;
		return () => (formDataRef.current = {});
	});

	/** Submit form and add new user data to db. */

	const handleSubmit = useCallback(
		async (e) => {
			e.preventDefault();
			let profileData = {
				firstName: formData.firstName,
				lastName: formData.lastName,
				email: formData.email,
				profileUrl: formData.profileUrl,
				password: formData.password
			};

			let updatedUser;
			let username = currentUser.username;
			try {
				updatedUser = await BoogieBotApi.saveProfile(username, profileData);
				history.push("/");
			} catch (err) {
				alert("Password is incorrect, please try again.");
				console.error(formErrors);
				return;
			}
			setFormData((form) => ({ ...form, password: "" }));
			setFormSuccess(true);
			setFormErrors([]);
			// trigger reloading of user information throughout the site
			setCurrentUser(updatedUser);
		},
		[ formData, history, currentUser.username, formErrors, setCurrentUser ]
	);

	/** Handle form data changing */

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((formData) => ({
			...formData,
			[name]: value
		}));
	};

	return (
		<div className="Form">
			<div className="Form-left">
				<span className="Form-update">{staticMsg.FORM_TIP}</span>
				<p>{staticMsg.EDIT_PROFILE_FORM_TIP_DESC}</p>
			</div>
			<div className="Form-right">
				<img className="Form-profile-pic" src={profile} alt="userprofilepicture" />
				<p>
					<b>{currentUser.username}</b>
				</p>

				<Form className="Form-body" onSubmit={handleSubmit}>
					<Row>
						<Col md={6}>
							<FormGroup>
								<Label htmlFor="firstName">{staticMsg.FORM_FIRST_NAME}</Label>
								<Input
									className="Form-input"
									name="firstName"
									id="firstName"
									value={formData.firstName}
									onChange={handleChange}
								/>
							</FormGroup>
						</Col>
						<Col md={6}>
							<FormGroup>
								<Label htmlFor="lastName">{staticMsg.FORM_LAST_NAME}</Label>
								<Input
									className="Form-input"
									name="lastName"
									id="lastName"
									value={formData.lastName}
									onChange={handleChange}
								/>
							</FormGroup>
						</Col>
					</Row>
					<FormGroup>
						<Label htmlFor="email">{staticMsg.FORM_EMAIL}</Label>
						<Input
							className="Form-input"
							name="email"
							id="email"
							value={formData.email}
							onChange={handleChange}
						/>
					</FormGroup>
					<FormGroup>
						<Label htmlFor="profileUrl">{staticMsg.FORM_PROFILE_URL}</Label>
						<Input
							className="Form-input"
							name="profileUrl"
							id="profileUrl"
							value={formData.profileUrl}
							onChange={handleChange}
						/>
					</FormGroup>
					<FormGroup>
						<Label htmlFor="password">{staticMsg.EDIT_PROFILE_FORM_CONFIRM_PWD}</Label>
						<Input
							className="Form-input"
							type="password"
							name="password"
							id="password"
							value={formData.password}
							onChange={handleChange}
						/>
					</FormGroup>

					{formSuccess ? (
						<Alert type="success" messages={[ "Updated successfully." ]} />
					) : null}
					<Button className="Form-btn btn-success">Save</Button>
				</Form>
			</div>
		</div>
	);
};

export default EditProfileForm;
