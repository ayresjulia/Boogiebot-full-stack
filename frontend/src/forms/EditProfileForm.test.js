import { render } from "@testing-library/react";
import React from "react";
import EditProfileForm from "./EditProfileForm";
import { UserProvider } from "../helpers/testUtils";

test("it renders without crashing", () => {
	render(
		<UserProvider>
			<EditProfileForm />
		</UserProvider>
	);
});

it("matches snapshot", () => {
	const { asFragment } = render(
		<UserProvider>
			<EditProfileForm />
		</UserProvider>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("renders a field in the form", () => {
	const { getByText } = render(
		<UserProvider>
			<EditProfileForm />
		</UserProvider>
	);
	expect(getByText("Confirm password to make changes")).toBeInTheDocument();
});
