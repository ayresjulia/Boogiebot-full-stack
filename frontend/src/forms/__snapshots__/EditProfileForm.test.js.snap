// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Form"
  >
    <div
      class="Form-left"
    >
      <span
        class="Form-update"
      >
        tip.
      </span>
      <p>
        while updating your info, keep in mind, that picking a great playlist is key for any party!
      </p>
    </div>
    <div
      class="Form-right"
    >
      <img
        alt="userprofilepicture"
        class="Form-profile-pic"
      />
      <p>
        <b>
          testuser
        </b>
      </p>
      <form
        class="Form-body"
      >
        <div
          class="row"
        >
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label
                class=""
                for="firstName"
              >
                First Name
              </label>
              <input
                class="Form-input form-control"
                id="firstName"
                name="firstName"
                type="text"
                value="testfirst"
              />
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label
                class=""
                for="lastName"
              >
                Last Name
              </label>
              <input
                class="Form-input form-control"
                id="lastName"
                name="lastName"
                type="text"
                value="testlast"
              />
            </div>
          </div>
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="email"
          >
            Email
          </label>
          <input
            class="Form-input form-control"
            id="email"
            name="email"
            type="text"
            value="test@test.net"
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="profileUrl"
          >
            Profile URL
          </label>
          <input
            class="Form-input form-control"
            id="profileUrl"
            name="profileUrl"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="password"
          >
            Confirm password to make changes
          </label>
          <input
            class="Form-input form-control"
            id="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <button
          class="Form-btn btn-success btn btn-secondary"
        >
          Save
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
